### PR TITLE
Support dependencies to go_library rules in the same repository

### PR DESCRIPTION
**Pull request by yugui**
* Extracted from #51 
* Depends on #57

### CODE REVIEW

File: go/tools/gazelle/gazelle/main.go [hunk 1]
```diff
@@ -74,10 +75,16 @@ func main() {
 	flag.Usage = usage
 	flag.Parse()
 
+	if *goPrefix == "" {
+		// TODO(yugui): Extract go_prefix from the top level BUILD file if
+		// exists
```
Comment (pmbethe09) [inline go/tools/gazelle/gazelle/main.go]: this comment can stay on the previous line (no line length limits)

File: go/tools/gazelle/rules/generator.go [hunk 2]
```diff
@@ -35,28 +36,70 @@ type Generator interface {
...
+	r, err := g.generate(rel, pkg)
+	if err != nil {
+		return nil, err
+	}
+	return []*bzl.Rule{r}, nil
+}
+
+func (g *generator) generate(rel string, pkg *build.Package) (*bzl.Rule, error) {
 	kind := "go_library"
 	name := "go_default_library"
```
Comment (pmbethe09) [inline go/tools/gazelle/rules/generator.go]: I might pull the "go_default_library" up to a constant with a comment about this expectation.

File: go/tools/gazelle/rules/resolve.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+	repo, pkg, name string
+	relative        bool
+}
+
+func (l label) String() string {
+	if l.relative {
+		return fmt.Sprintf(":%s", l.name)
+	}
+	if l.repo != "" {
+		return fmt.Sprintf("@%s//%s:%s", l.repo, l.pkg, l.name)
```
Comment (pmbethe09) [inline go/tools/gazelle/rules/resolve.go]: here and below; "@repo//lib:lib" should just be "@repo//lib"

**Comment by pmbethe09** [comment]
LGTM with a few comments.

File: go/tools/gazelle/gazelle/main.go (hunk 1 above)
Comment (yugui) [inline go/tools/gazelle/gazelle/main.go]: https://golang.org/wiki/CodeReviewComments#line-length

> Comments are typically wrapped before no more than 80 characters

File: go/tools/gazelle/rules/generator.go (hunk 2 above)
Comment (yugui) [inline go/tools/gazelle/rules/generator.go]: done

File: go/tools/gazelle/rules/resolve.go (hunk 3 above)
Comment (yugui) [inline go/tools/gazelle/rules/resolve.go]: done

**Pull request merged**